### ISSUE TITLE
`create_segment_table` calculation "errors"

### ISSUE THREAD

**Issue by KennyChenBasis** [mjishnu/pypdl#13]
Hi,

I was reviewing this project's code and came across what looks to be some distasteful code in `create_segment_table`. In
```python
        if segment != (segments - 1):
            end -= 1  # [0-100, 100-200] -> [0-99, 100-200]
        # No segment_size+=1 for last setgment since final byte is end byte
```
`end -= 1` should also be applied to the last segment; if the size is `200`, the last byte is at index `199` (0-index, inclusive).

There is also careless use of floating points -- `int(size / segments * segments)` doesn't always equal `size`, e.g. `int(109 / 11 * 11)` returns `108`. This problem is accidentally mitigated by the above problem. However, cases where the calculation is `2` (or greater) less can also occur, though you'll need very large files: `int(9999999999999966 / 17 * 17)` returns `9999999999999964`.

Setting the last segment's `end` as `size - 1` is acceptable, though the sane thing to do is to also have `partition_size` be an int.

**Comment by mjishnu** [comment]
thanks for reviewing, your analysis is correct. when the size is large it would result in incomplete segments. I will fix it or if you are willing please do create a PR.

**Comment by mjishnu** [comment]
fixed you can check [f09ae1b](https://github.com/mjishnu/pypdl/commit/f09ae1bec90edd6ac683a0399392c640c74816c2). do tell me if you find any other bugs.

**Comment by KennyChenBasis** [comment]
LGTM
